flyteadmin/pkg/async/shared.go

```diff
@@ -10,6 +11,24 @@ import (
 // RetryDelay indicates how long to wait between restarting a subscriber connection in the case of network failures.
 var RetryDelay = 30 * time.Second
 
+func RetryOnSpecificErrorCodes(totalAttempts int, delay time.Duration, f func() (*http.Response, error), IsErrorCodeRetryable func(*http.Response) bool) error {
```

Comment:
Can we instead reuse this one which comes from K8s utils. We are using this in flyteadmin already https://github.com/flyteorg/flyte/blob/master/flyteadmin/scheduler/executor/executor_impl.go#L100

Reply:
Definitely can, is there any specific reason as to why you're wanting to use this one? We use the other retry functions in [shared.go](https://github.com/squiishyy/flyte/blob/03fc38a814552d869f67f2a9bd2bf2b0c3a49979/flyteadmin/pkg/async/shared.go) elsewhere in FlyteAdmin as well

Reply:
Oh. For consitency across our repos , would be good to use one if they provide the same functionality. 
In cloud repo too we use the k8s retry util in grafana dataplane with usage of RetryOnConflict and hence was seeing if we could use the same

Reply:
Ah gotcha! makes sense, happy to migrate over to that :) will work on that now